src/data/mv_polynomial.lean

```diff
@@ -423,6 +423,12 @@ instance eval₂.is_semiring_hom : is_semiring_hom (eval₂ f g) :=
   map_one := eval₂_one _ _,
   map_add := λ p q, eval₂_add _ _,
   map_mul := λ p q, eval₂_mul _ _ }
+
+/-- `mv_polynomial.eval₂` as a `ring_hom`. -/
+def eval₂_hom (f : α →+* β) (g : σ → β) : mv_polynomial σ α →+* β := ring_hom.of (eval₂ f g)
+
+@[simp] lemma coe_eval₂_hom (f : α →+* β) (g : σ → β) : ⇑(eval₂_hom f g) = eval₂ f g := rfl
```

Comment:
Would it be good to make this a very low priority simp-lemma? So that `simp` gets the chance to use the simp-lemmas `ring_hom.map_add` etc first?

Reply:
I hope that `eval₂_hom` is a temporary workaround: Lean 3 can coerce it to a function, and the only reason not to define it as the main interface is that the proof of `eval₂_mul` is not cheap. Once we move `algebra` below `monoid_algebra` in the `import` chain and prove that any `monoid_hom` to an algebra over `R` generates an algebra homomorphism, it can go away.